### ISSUE TITLE
 Error: Sync Storage `init()` needs to be called before using it.

### ISSUE THREAD

**Issue by felixchan** [RevenueCat/react-native-purchases#278]
```
ExceptionsManager.js:180 Error: Sync Storage `init()` needs to be called before using it.
    at SyncStorage.checkIfLoaded (storage.native.js:77)
    at SyncStorage.setItem (storage.native.js:32)
    at updateState (use-state-persist.js:35)
    at checkPurchase$ (Home.js:115)
    at tryCatch (runtime.js:63)
    at Generator.invoke [as _invoke] (runtime.js:293)
    at Generator.next (runtime.js:118)
    at tryCatch (runtime.js:63)
    at invoke (runtime.js:154)
    at runtime.js:164
```

The error above is caused by `getPurchaserInfo`.

```
export const getPurchase = async () => {
    try {

        Purchases.setup('XXX') //I've already done this in parent, but doing it again, just in case.
        const purchaserInfo = await Purchases.getPurchaserInfo()
        if (Object.entries(purchaserInfo.entitlements.active).length) {
            return Promise.resolve(purchaserInfo)
        }
        return Promise.resolve(null)
    } catch (e) {
        return Promise.resolve(null)
    }
}
```

```
const Home = (props) => {
    const [customer, setCustomer] = useState()
    const checkPurchase = async () => {
        const purchase = await getPurchase()
        setCustomer(purchase)
    }
    useEffect(() => {
        checkPurchase()
    }, [])
    if (typeof customer === 'undefined') return <View />
    if (customer === null) return <Splash {...props} checkPurchase={checkPurchase} />
    return <Access {...props} />
}
```

**Comment by aboedo** [comment]
Hi! Thanks for reporting! 
A few thoughts and questions: 
- This is very likely unrelated, but as a general thing, make sure `setup` gets called only once in the lifetime of your app. Configuring multiple times can lead to unexpected behavior. Again, this is most likely unrelated, but still worth pointing out. 
- Is this happening on iOS or Android? 
- The error message seems to suggest the app is using `Sync Storage`, but I don't believe our SDK uses it. Could this error be coming from the `setCustomer(purchase)` line?

**Comment by felixchan** [comment]
Thank you Aboedo. It seems to be an issue on my side. Apologies.

Your product is wonderful. Love the straightforward approach.

**Comment by aboedo** [comment]
@felixchan no worries! thanks for reporting back! 

> Your product is wonderful. Love the straightforward approach.

You just made my day 😃😃😃
